lib/fluent/plugin/in_forward.rb

```diff
@@ -82,11 +83,15 @@ def shutdown
       @usock.close
       @thread.join
       @lsock.close
+      @connections.each { |c| c.close }
     end
 
     def listen
       log.info "listening fluent socket on #{@bind}:#{@port}"
-      s = Coolio::TCPServer.new(@bind, @port, Handler, @linger_timeout, log, method(:on_message))
+      s = Coolio::TCPServer.new(@bind, @port, Handler, @linger_timeout, log, method(:on_message)) do |connection|
+        @connections.reject! { |conn| conn.closed? }
```

Comment:
I think it doesn't need to be guarded.
(I would like to confirm just in case; does `this block` mean `do |connection| ... end`, not `{ |conn| conn.closed? }`, right?)

`@connections` is used for closing accepted connections at shutdown, so already closed connections don't need to be included in. I think it is safe that several processes or threads come in this block at the same time.